### PR TITLE
Fix settings' strict CSP

### PR DESCRIPTION
**Pull request by ChristophWurst**
This will remove any runtime compilation (used to instantiate the settings component before), switch to a webpack devtool option that is CSP-compliant and removes the runtime compiler to decrease the overall bundle size. This is motived by the fact that compilation in the browser is now prevent anyway with the stricter CSP in Nextcloud 15.

Ref https://vuejs.org/v2/guide/installation.html#Runtime-Compiler-vs-Runtime-only.

### CODE REVIEW

File: settings/src/main-admin-security.js [hunk 1]
```diff
@@ -1,13 +1,10 @@
 import Vue from 'vue'
 
-import AdminTwoFactor from './components/AdminTwoFactor'
+import AdminTwoFactor from './components/AdminTwoFactor.vue'
+
+__webpack_nonce__ = btoa(OC.requestToken)
```
Comment (rullzer) [inline settings/src/main-admin-security.js]: we already set this in main.js is this then also needed here?

Comment (ChristophWurst) [reply settings/src/main-admin-security.js]: yes, these are two separate entries.

How about we actually set this globally? Then we won't have to adjust any app.

Comment (rullzer) [reply settings/src/main-admin-security.js]: If we can do that then yes please!

Comment (ChristophWurst) [reply settings/src/main-admin-security.js]: -> https://github.com/nextcloud/server/pull/11846

**Review by rullzer — APPROVED** [review]
Didn't do :boom: fine by me

**Review by MorrisJobke — APPROVED** [review]
Tested and works 👍

**Pull request merged**